### PR TITLE
Config requirement

### PR DESCRIPTION
**Pull request by ciliakas**
Implemented options pattern for email sender and removed redundant code in startup.

### CODE REVIEW

File: src/Startup.cs [hunk 1]
```diff
@@ -37,22 +37,10 @@ public void ConfigureServices(IServiceCollection services)
...
-            services.AddScoped(provider =>
-            {
-                var db = provider.GetService<PiggyDbContext>();
-                return new GoalDatabase(db);
-            });
-            services.AddScoped<EmailSender>();
+            services.AddScoped<UserDatabase>();
+            services.AddScoped<EntryDatabase>();
+            services.AddScoped<GoalDatabase>();
+            services.Configure<EmailSender>(options => Configuration.GetSection("Email").Bind(options));
```
Comment (Stokalas) [inline src/Startup.cs:44]: So he know which data to read as options by section? And then assigns it to same name properties?

File: src/appsettings.json [hunk 2]
```diff
@@ -6,5 +6,18 @@
...
+    "RecoveryMessageSubject": "Password Recovery",
+    "RecoveryMessageBody": "Your password recovery code is: ",
+    "GreetingMessageSubject": "Welcome to ePiggy!",
+    "GreetingMessageBody": "Thank you for joining ePiggy! Your wallet thanks you as well!",
+    "FarewellMessageSubject": "Farewell from the ePiggy team",
+    "FarewellMessageBody": "We're sorry to see you go, please leave your feedback of our app.",
+    "MaxRandValue": 999999,
+    "SmtpEmail": "smtp.gmail.com",
+    "Port": 587
+  },
```
Comment (Stokalas) [inline src/appsettings.json:21]: And as I understand when we read these as options we can change them during runtime and changes will be applied without restarting?

**Review by Stokalas — APPROVED** [review]
Seems well done. Nice to see that requirements are coming to an end bit by bit.
Left a few questions about how it works, as always :)

File: src/Startup.cs (hunk 1 above)
Comment (ciliakas) [reply src/Startup.cs:44]: Basically that's how it works, you can read more here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-5.0

File: src/appsettings.json (hunk 2 above)
Comment (ciliakas) [reply src/appsettings.json:21]: I think that's how it works, but I tried for a bit and couldn't get it to work, so maybe I'm misunderstanding it

**Pull request merged**